core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -176,7 +192,17 @@ public void validateIndexName(String index, ClusterState state) {
         }
     }
 
-    public void createIndex(final CreateIndexClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
+    public void createIndex(final CreateIndexClusterStateUpdateRequest request,
+                            final ActionListener<ClusterStateUpdateResponse> listener) {
+
+        final int numShardCopies = IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.get(request.settings()) + 1;
+        if (request.waitForActiveShards().exceeds(numShardCopies)) {
```

Comment:
@bleskes There would be some tradeoffs in moving the waiting to the `TransportCreateIndexAction`.  It will indeed simplify the code, but there are other actions other than `TransportCreateIndexAction` that call `MetaDataCreateIndexService#createIndex`, namely `TransportRolloverAction` and `TransportShrinkAction`.  It may be correct that neither of those two actions should do any waiting anyway.  But I was also thinking, now that our default wait_for_active_shards is 1, why not have all transport actions (TransportBulkAction, TransportUpdateAction, etc) just wait using the default (instead of specifying NONE) as it would simplify things?

The other tradeoff is, if the user specifies wait_for_active_shards > number_of_replicas + 1, here we can throw an error on index creation itself with a message that the request itself has something wrong in it.  If we move the waiting to the TransportCreateIndexAction, then we can only validate the wait_for_active_shards part of the request after the index has been created.